code/modules/projectiles/guns/misc/hygienegun.dm

```diff
@@ -0,0 +1,57 @@
+/obj/item/gun/hygienegun
+	name = "Stink Removal Device"
+	desc = "Cleans the stink right off any crew member."
+	icon = 'icons/obj/guns/projectile.dmi'
+	icon_state = "hygienegun"
+	w_class = WEIGHT_CLASS_SMALL
+	var/obj/item/pressurizedsoap/ammo
+
+/obj/item/gun/hygienegun/Initialize()
+	. = ..()
+	chambered = new /obj/item/ammo_casing/caseless/hygienepellet(src)
+
+/obj/item/gun/hygienegun/can_shoot()
+	return ammo.charges
+
+/obj/item/gun/hygienegun/process_chamber()
+	if(!ammo.charges)
+		return FALSE
+	if(chambered && !chambered.BB)
+		chambered.newshot()
+		ammo.charges--
+
+/obj/item/gun/hygienegun/update_icon()
+	if(ammo.charges)
+		add_overlay("hygienegun_clean")
+	else
+		cut_overlays()
+	
+
+/obj/item/gun/hygienegun/attackby(obj/item/A, mob/user, params)
```

Comment:
soap is craftable so would be cool if soap could be used to charge it as well

Reply:
there's also urinal cakes but those might be infinite

Reply:
ok i will add the behavior for soap